mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -5090,18 +5119,37 @@ static void affixThreadwiseCopyV2Attributes(miopen::ThreadwiseCopyV2Op top,
   }
 
   top->setAttr("bwd_padding_kernel_status", b.getI32IntegerAttr(status));
-  top->setAttr("dim_access_order", b.getArrayAttr({
-                                       b.getI32IntegerAttr(0),
-                                       b.getI32IntegerAttr(1),
-                                       b.getI32IntegerAttr(2),
-                                       b.getI32IntegerAttr(3),
-                                       b.getI32IntegerAttr(4),
-                                   }));
+  if (isSwizzled) {
+    top->setAttr("dim_access_order", b.getI32ArrayAttr({0, 1, 2, 3, 4, 5}));
+  } else {
+    top->setAttr("dim_access_order", b.getI32ArrayAttr({0, 1, 2, 3, 4}));
+  }
+
+  // Account for split m/n dimension
+  bool vectorStoreOverride = false;
+  int64_t vectorGemmDim =
+      gop->getAttrOfType<IntegerAttr>("matrix_c_source_vector_read_dim")
+          .getInt();
+  if (vectorGemmDim == 1) {
```

Comment:
some comments might be needed here to explain the rationale.
or perhaps introduce an enumeration to hide these numbers?